src/Models/AbstractEntity.php

```diff
@@ -165,15 +160,89 @@ abstract public function create(
         ?int $customId = null,
         ?string $metadata = null,
         int $rating = 0,
-        ?int $contentType = null,
-        bool $forceExpTpl = false,
-        string $defaultTemplateHtml = '',
-        string $defaultTemplateMd = '',
-        ?string $color = null,
+        BodyContentType $contentType = BodyContentType::Html,
     ): int;
 
     abstract public function duplicate(bool $copyFiles = false, bool $linkToOriginal = false): int;
 
+    public function createFromTemplate(int $templateId, ?string $title = null): int
+    {
+        $TemplateType = $this->entityType->toTemplateType($this->Users, $templateId);
+        $template = $TemplateType->readOne();
+        $id = $this->create(
+            title: $title ?? $template['title'],
+            body: $template['body'],
+            canread: $template['canread_target'],
+            canwrite: $template['canwrite_target'],
+            canreadIsImmutable: (bool) $template['canread_is_immutable'],
+            canwriteIsImmutable: (bool) $template['canwrite_is_immutable'],
+            category: $template['category'],
+            status: $template['status'],
+            metadata: $template['metadata'],
+            rating: $template['rating'],
+            contentType: BodyContentType::from($template['content_type']),
+        );
+        $tags = array_column($TemplateType->Tags->readAll(), 'tag');
+        $this->ItemsLinks->duplicate($template['id'], $id, true);
+        $this->ExperimentsLinks->duplicate($template['id'], $id, true);
+        $CompoundsLinks = LinksFactory::getCompoundsLinks($this);
+        $CompoundsLinks->duplicate($templateId, $id, fromItemsTypes: true);
+        $this->Steps->duplicate($template['id'], $id, true);
```

Comment:
```suggestion
        $this->ItemsLinks->duplicate($templateId, $id, true);
        $this->ExperimentsLinks->duplicate($templateId, $id, true);
        $CompoundsLinks = LinksFactory::getCompoundsLinks($this);
        $CompoundsLinks->duplicate($templateId, $id, fromItemsTypes: true);
        $this->Steps->duplicate($templateId, $id, true);
```
It seems they're the same
Or use `$template['id']` everywhere instead, since it's the result of the readOne()